### DISCUSSION TITLE
Release binaries

### DISCUSSION THREAD

**Discussion by josephzidell** [josephzidell/joelang#31] [Q&A]
Since this is my first time writing a compiler, this is a venture on new terrain. Turns out writing a compiler isn't as straight-forward as I thought :smile:

Here is an outline the steps I took, and if anyone is able to point out mistakes or better options, all feedback is appreciated. :pray: 

### Similar players:
Since [Rust uses LLVM](https://rustc-dev-guide.rust-lang.org/overview.html#code-generation), that seemed like a decent place to begin. The current Joelang compiler code is written in TS, and the most relevant NPM package I could find was https://www.npmjs.com/package/llvm-bindings. This required LLVM to be pre-installed, which I took notice of, didn't like, but bit that bullet.

With LLVM installed manually, the `.o` object file [is being generated](https://github.com/josephzidell/joelang/blob/v0.0.1/compile.ts#L333-L342), followed by [the executable](https://github.com/josephzidell/joelang/blob/v0.0.1/compile.ts#L345). And that worked. Well, sort of.

I made a mistake. During testing the executable was generated in the Joelang project root, and it ran and compiled Joelang source code. Fantastic. However, turns out it was relying on certain files in the `node_modules/` directory. If you move the executable anywhere else, you get an error similar to
```
Error: Could not locate the bindings file
```

Furthermore, even if LLVM is installed, it may be possible that it doesn't include the specific command we're currently using: `llc`

I'm currently researching alternative solutions for this. If anyone knows how to include the LLVM binaries in the executable, that would be best. That would allow the Getting Started process to be easier, with the goal of a download-and-run with no prerequisites.

**Comment by Person-93** [comment]
> Fantastic. However, turns out it was relying on certain files in the node_modules/ directory. If you move the executable anywhere else, you get an error similar to

It's worse than that. If your _current directory_ is anything but the project root, it fails.

**Reply by Person-93** [reply]
Never mind. Seems this was fixed. I'm actually getting a different error.
